### ISSUE TITLE
StatsModels.jl >v0.7 changes order of parameters

### ISSUE THREAD

**Issue by sindresops** [JuliaStats/StatsModels.jl#286]
My code was breaking because StatsModels.jl has changed the default ordering of FormulaTerm. (Intercept) used to come first, now its last. I was dependent on GLM.coef() returning the regression coefficients in that specific order. The new change is good, since now the higher order terms come first. But just an FYI in case anyone else experiences the same.

Edit 1: (Added MWE)

`using Pkg;
using DataFrames;
Pkg.add(name="StatsModels",version="0.6.33");
using GLM;
x = collect(1:10); y = 2x .+ randn(length(x));
lm(@formula(y~x+1),DataFrame(x=x,y=y));`

Returns 
![image](https://user-images.githubusercontent.com/29738855/229708102-587db218-0657-466b-b801-c00f808cc732.png)

`using Pkg;
using DataFrames;
Pkg.add(name="StatsModels",version="0.7");
using GLM;
x = collect(1:10); y = 2x .+ randn(length(x));
lm(@formula(y~x+1),DataFrame(x=x,y=y));`

Returns 
![image](https://user-images.githubusercontent.com/29738855/229708143-380ed8a1-d761-4baa-8a50-47c40592436e.png)

Other info
Julia version 1.8.0
GLM v1.8.2

**Comment by kleinschmidt** [comment]
I think the issue is that the constant term is incorrectly assigned the same degree as `x`.  Sorting works correctly with an interaction term like this:

```julia
julia> f2 = @formula(y ~ x & z + x + 1)
FormulaTerm
Response:
  y(unknown)
Predictors:
  x(unknown)
  1
  x(unknown) & z(unknown)

(jl_QKrdLJ) pkg> st
Status `/private/var/folders/kg/y0c0ksr56_g800hjvcpx_d4w0000gp/T/jl_QKrdLJ/Project.toml`
  [a93c6f00] DataFrames v1.5.0
  [38e38edf] GLM v1.8.2
  [3eaba693] StatsModels v0.7.0
```

**Comment by kleinschmidt** [comment]
@sindresops this is fixed on master now and will be released as StatsModels 0.7.1: https://github.com/JuliaRegistries/General/pull/81005

Thanks for the report!

**Comment by sindresops** [comment]
Thanks for patching!
